haiku-games/bong/bong-1.4~git.recipe

```diff
@@ -29,18 +30,26 @@ BUILD_PREREQUIRES="
 	cmd:make
 	"
 
+GLOBAL_WRITABLE_FILES="
+	settings/bong auto-merge
+	"
+
 USER_SETTINGS_FILES="
-	settings/bong
+	settings/bong template data/bong/defaultconfig
 	"
 
 BUILD()
 {
 	make $jobArgs OBJ_DIR=objects MACHINE=BePC CC=g++ LD=g++
+	mkdir -p settings
+	echo "my bong prefs" >settings/bong
 }
 
 INSTALL()
 {
-	mkdir -p $appsDir
-	cp -a objects/bong $appsDir
-	addAppDeskbarSymlink $appsDir/bong
+	install -d $appDir $dataDir/bong $settingsDir
+	install -t $appsDir objects/bong
+	install -t $settingsDir settings/bong
+	install settings/bong $dataDir/bong $dataDir/bong/defaultconfig
```

Comment:
I think you forgot to remove the `$dataDir/bong` in this line.
```diff
-	install settings/bong $dataDir/bong $dataDir/bong/defaultconfig
+	install settings/bong $dataDir/bong/defaultconfig
```
BTW, does the program have settings?
If not, then you can simply drop the GLOBAL_WRITABLE_FILES, the USER_SETTINGS_FILES, and 2 last calls to install in INSTALL.

Reply:
Yes, it does have settings.
https://github.com/HaikuArchives/Bong/blob/906d822b79bdae9328fe743dffe2871f96a30e0e/README.txt#L48
So i'll change all `settings/bong` to `settings/Bong_prefs`

But it's not the problem. I don't know what am i missing but the packaging progress stucked. It said:

` install: failed to access '/packages/bong-1.4~git-1/.self/apps': No such file or directory
`

When i check the package directory, it still use old name

```
~/haikuports/haiku-games/bong> ls -l work-1.4~git/packages/
total 0
...
dr-xr-xr-x 1 user root 0 Dec 23 01:58 bong-1.0.0-1
...
```

Reply:
Oops, there's a typo in line 50:
```diff
-	install -d $appDir $dataDir/bong $settingsDir
+	install -d $appsDir $dataDir/bong $settingsDir
```
(Should be app**s**Dir)

Regarding the prefs, if there are several files, then it is recommended to create a directory in $settingsDir, say, bong, and put all the preferences files in there. If there is only one file and if its name can be clearly associated to the app, then that's OK to put it directly in $settingsDir.
BTW, in case you create $settingsDir, you need to declare something that would have this kind of structure:

```
USER_SETTINGS_FILES="
	settings/bong directory
	settings/bong/screen template data/bong/defaultsettings/screen
	settings/bong/keys template data/bong/defaultsettings/keys
	settings/bong/high-scores template data/bong/defaultsettings/high-scores
	"
```
and ajust INSTALL to put these default settings in both `$settingsDir/bong/` and `$dataDir/bon/defaultsettings/`.
Of course the names here are just examples.

Reply:
It worked, thanks for the help @fbrosson